system/CodeIgniter.php

```diff
@@ -407,6 +407,8 @@ protected function handleRequest(RouteCollectionInterface $routes = null, Cache
 			{
 				return $returnResponse ? $possibleResponse : $possibleResponse->pretend($this->useSafeOutput)->send();
 			}
+
+			$this->request = $possibleResponse;
```

Comment:
We need to check the actual return type. `Filters::run()` has the following behavior:
1. If a Filter returns a `RequestInterface` then store it and continue
2. If a Filter returns a `ResponseInterface` then return it and quit (handled just above)
3. If a Filter returns something `empty()` then keep running filters
4. If a Filter returns anything else return it

Case 4 could cause us issues, since `$this->request` needs to be "the current Request object". So something like:

```suggestion
			if ($possibleResponse instanceof RequestInterface)
			{
				$this->request = $possibleResponse;
			}
```

Reply:
I missed that 4th case so that makes sense. I have pushed up that change.